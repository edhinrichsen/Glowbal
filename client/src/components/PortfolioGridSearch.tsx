import React, { useState } from 'react';

import PortfolioObjectSearch from './PortfolioObjectSearch';

import './PortfolioGrid.css';

type PortfolioGridSearchProps = {
  data: Array<PortfolioObjectSearchData>;
};

type PortfolioObjectSearchData = {
  // routes set in snake_case
  project_id: string;
  project_name: string;
  author_first_name: string;
  author_last_name: string;
  user_id: string;
};

function PortfolioGridSearch(props: PortfolioGridSearchProps) {
  const [portfolioObjectOpen, setPortfolioObjectOpen] = useState(false);

  const openPortfolioObject = (open: boolean) => {
    setPortfolioObjectOpen(open);
  };

  console.log(props.data);

  if (Object.keys(props.data).length === 0) {
    return <></>;
  } else {
    return (
      <div className={`${portfolioObjectOpen ? 'portfolio-object-open' : ''}`}>
        <div className="grid">
          {props.data.map((portfolioData, index) => {
            return (
              <PortfolioObjectSearch
                key={portfolioData.project_id}
                data={portfolioData}
                portfolioObjectOpen={openPortfolioObject}
              />
            );
          })}
        </div>
      </div>
    );
  }
}

export default PortfolioGridSearch;
