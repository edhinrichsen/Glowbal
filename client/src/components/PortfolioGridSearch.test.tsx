import React from 'react';
import { shallow } from 'enzyme';
import PortfolioGridSearch from './PortfolioGridSearch';

describe('Testing PortfolioGridSearch', () => {
  it('renders home without crashing', () => {
    const component = shallow(
      <PortfolioGridSearch
        data={[
          {
            project_id: '58',
            project_name: 'Cards',
            author_first_name: 'Alex',
            author_last_name: 'Shepard',
            user_id: '5f954a93aeac5f006f9750c2',
          },
          {
            project_id: '53',
            project_name: 'Breed App',
            author_first_name: 'Alex',
            author_last_name: 'Shepard',
            user_id: '5f954a93aeac5f006f9750c2',
          },
        ]}
      />
    );
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
