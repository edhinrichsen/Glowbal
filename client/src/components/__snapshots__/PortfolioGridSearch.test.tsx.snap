// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PortfolioGridSearch renders home without crashing 1`] = `
<div
  className=""
>
  <div
    className="grid"
  >
    <PortfolioObjectSearch
      data={
        Object {
          "author_first_name": "Alex",
          "author_last_name": "Shepard",
          "project_id": "58",
          "project_name": "Cards",
          "user_id": "5f954a93aeac5f006f9750c2",
        }
      }
      key="58"
      portfolioObjectOpen={[Function]}
    />
    <PortfolioObjectSearch
      data={
        Object {
          "author_first_name": "Alex",
          "author_last_name": "Shepard",
          "project_id": "53",
          "project_name": "Breed App",
          "user_id": "5f954a93aeac5f006f9750c2",
        }
      }
      key="53"
      portfolioObjectOpen={[Function]}
    />
  </div>
</div>
`;
